extensions/amp-story-interactive/0.1/amp-story-interactive-results-detailed.js

```diff
@@ -36,7 +61,80 @@ export class AmpStoryInteractiveResultsDetailed extends AmpStoryInteractive {
 
   /** @override */
   buildComponent() {
-    this.rootEl_ = htmlFor(this.element)`<p>Detailed results component</p>`;
+    this.rootEl_ = buildResultsDetailedTemplate(this.element);
+    this.buildTop();
     return this.rootEl_;
   }
+
+  /** @override */
+  onInteractiveReactStateUpdate(interactiveState) {
+    const components = Object.values(interactiveState);
+    if (this.selectedResultEls_) {
+      // Function not passed in directly to ensure "this" works correctly
+      components.forEach((e) => this.updateSelectedResult_(e));
+    } else {
+      this.initializeSelectedResultContainers_(components);
```

Comment:
It makes sense for this to work this way if the component was built after all the other components. However, that's not guaranteed (eg: if you reload a story when you're looking at the results component, it can load before the quizzes on page 1). So, we could take the approach of receiving the events, and recreate the layout if the number of quizzes or polls changes. It will work in most cases just like what you have now, but for the cases where new quizzes or polls get loaded after the initializer, they would get added to the component (moving or recreating the existing bubbles)